vulture_os/toolkit/api_parser/cisco_umbrella/cisco_umbrella.py

```diff
@@ -0,0 +1,173 @@
+
+    def _connect(self):
+        try:
+            # Check for expiration with 10 seconds difference to be sure token will still be valid for some time
+            if not self.cisco_umbrella_access_token or not self.cisco_umbrella_expires_at or (self.cisco_umbrella_expires_at - timedelta(seconds=10)) < timezone.now():
+                self._get_token()
+
+            if self.session is None:
+                self.session = requests.Session()
+
+                headers = self.HEADERS
+                headers.update({'Authorization': f"Bearer {self.cisco_umbrella_access_token}"})
+
+                self.session.headers.update(headers)
+
+        except Exception as err:
+            raise CiscoUmbrellaAPIError(err)
+
+    def __execute_query(self, method, url, query, timeout=30):
+        '''
+        raw request doesn't handle the pagination natively
+        '''
+        self._connect()
+
+        response = self.session.get(url,
+            params=query,
+            headers=self.HEADERS,
+            timeout=timeout,
+            proxies=self.proxies,
+            verify=self.api_parser_custom_certificate if self.api_parser_custom_certificate else self.api_parser_verify_ssl
+        )
+
+        # response.raise_for_status()
+        if response.status_code != 200:
+            raise CiscoUmbrellaAPIError(f"Error at Cisco-Umbrella API Call URL: {url} Code: {response.status_code} Content: {response.content}")
+
+        return response.json()
+
+    def test(self):
+        try:
+            result = self.get_logs(timezone.now()-timedelta(hours=2), timezone.now())
+
+            return {
+                "status": True,
+                "data": result
+            }
+        except Exception as e:
+            logger.exception(f"[{__parser__}]:test: {e}", extra={'frontend': str(self.frontend)})
+            return {
+                "status": False,
+                "error": str(e)
+            }
+
+    def get_logs(self, since, to, index=0):
+
+        payload = {
+            'offset': index,
+            'limit': 5000,
+            'from': int(since.timestamp() * 1000),
+            'to': int(to.timestamp() * 1000),
+        }
+        logger.debug(f"[{__parser__}]:get_alerts: Cisco-Umbrella query parameters : {payload}",
+                     extra={'frontend': str(self.frontend)})
+        return self.__execute_query("GET", self.ACTIVITY_URL, payload)
+
+    def format_log(self, log):
+        return json.dumps(log)
+
+    def execute(self):
+        since = self.last_api_call or (timezone.now() - timedelta(minutes=15))
+        to = min(timezone.now(), since + timedelta(minutes=15))
+
+        msg = f"Parser starting from {since} to {to}."
+        logger.info(f"[{__parser__}]:execute: {msg}", extra={'frontend': str(self.frontend)})
+
+        index = 0
+        logs_count = self.LIMIT_MAX
+        while logs_count == self.LIMIT_MAX and index <= self.OFFSET_MAX:
```

Comment:
What happens to the remaining logs in that time period if there are more than 10k logs?